specification/communication/data-plane/Chat/preview/2022-11-15-preview8/communicationserviceschat.json

```diff
@@ -0,0 +1,1977 @@
+            "description": "Too many requests.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "503": {
+            "description": "Service unavailable.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-examples": {
+          "Delete chat thread": {
+            "$ref": "./examples/Threads_DeleteChatThread.json"
+          }
+        }
+      }
+    },
+    "/chat/threads/{chatThreadId}/typing": {
+      "post": {
+        "tags": [
+          "TypingNotifications"
+        ],
+        "summary": "Posts a typing event to a thread, on behalf of a user.",
+        "operationId": "ChatThread_SendTypingNotification",
+        "consumes": [
+          "application/json",
+          "application/merge-patch+json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "chatThreadId",
+            "description": "Id of the thread.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "sendTypingNotificationRequest",
+            "description": "Details of the typing notification request.",
+            "schema": {
+              "$ref": "#/definitions/SendTypingNotificationRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful."
+          },
+          "401": {
+            "description": "Unauthorized.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "403": {
+            "description": "Forbidden.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "429": {
+            "description": "Too many requests.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "503": {
+            "description": "Service unavailable.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-examples": {
+          "Post typing event to a thread": {
+            "$ref": "./examples/Messages_SendTypingNotification.json"
+          }
+        }
+      }
+    },
+    "/chat/threads/{chatThreadId}/{chatMessageId}/teamsinterop/image/{imageId}/$value": {
```

Comment:
This misses the `/messages/` part after the {chatThreadId}.
`/image` should be plural `/images` because it's a collection
If we stick with `teaminterop` it should be camelCase `teamsInterop`. We haven't used the term "interop" anywhere in our APIs so far, however, so we can debate a bit more.

Reply:
The `$value` at the end can be dropped. I had noted it previously because the path that was discussed was attachments, not images

Reply:
After we've cleaned up the route, we should ask if we need to document it in the swagger or just rely on the urls in the attachment to be opaque to the client